src/main/python/inverse_parser/_node_restorer.py

```diff
@@ -0,0 +1,684 @@
+# Copyright (c) Aniskov N.
+
+import ast
+import xml.etree.ElementTree as ET
+from typing import *
+
+from src.main.python.inverse_parser._merged_types_restorer import _MergedTypesRestorer
+from src.main.python.inverse_parser._attribute_setter import _AttributeSetter
+from src.main.python.inverse_parser._xml_node_children_getter import _XmlNodeChildrenGetter
+
+
+class _NodeRestorer:
+
+    @staticmethod
+    def restore(xml_node: ET.Element) -> ast.AST:
+        """
+        Partially restores python AST from XML AST recursively
+        :param xml_node: AST node in xml format to restore
+        :return: partially restored python AST
+        """
+
+        (py_node_type, merged_types) = _MergedTypesRestorer.get_types_from_tag(xml_node)
+        py_node = py_node_type()
+        _MergedTypesRestorer.set_ctx_and_ops(py_node, merged_types)
+        if not py_node_type == ast.Module and not py_node_type == type(None):  # ast.Module doesn't need to be localized
+            _NodeRestorer.localize_node(xml_node, py_node)
+
+        try:
+            restore_py_node = _NodeRestorer.py_node_type_to_restorer[py_node_type]
+
+        except KeyError:
+            raise RuntimeError(f'Unknown node type: {py_node_type}.')
+
+        restore_py_node(xml_node, py_node)
+        return py_node
+
+    # Node restorers:
+
+    # - Literals:
+
+    @staticmethod
+    def restore_ast_constant(xml_node: ET.Element, py_node: ast.AST) -> None:
+        _AttributeSetter.set_const_value(xml_node, py_node, py_node_attrib_name='value')
```

Comment:
You often use different string constants like 'NoneType' or 'value'. Maybe should you add an enum class to store these names and use them? Because it is too easy to make a mistake when you use it in this way like now. What do you think?

Reply:
I think this will be a little superfluous and will not increase the readability of the code.